From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: chickeneer <emcchickeneer@gmail.com>
Date: Tue, 30 Dec 2025 11:53:22 -0600
Subject: [PATCH] Better handle super old legacy data.

Add priority value to old saved ticks

Very old versions of minecraft did not have a priority for saved ticks
This injects the default priority level for loading in those ancient
ticks

Also remove the old Owner from tamed mobs (wolves) when moving the UUID
field to that datavalue

diff --git a/ca/spottedleaf/dataconverter/minecraft/versions/V2514.java b/ca/spottedleaf/dataconverter/minecraft/versions/V2514.java
index 5225a29c236507b5dc682b06679bb82a49e84068..b783288723ec87d1ae907cf9ae7ab594dd675aad 100644
--- a/ca/spottedleaf/dataconverter/minecraft/versions/V2514.java
+++ b/ca/spottedleaf/dataconverter/minecraft/versions/V2514.java
@@ -239,7 +239,7 @@ public final class V2514 {
 
     private static void updateAnimalOwner(final MapType data) {
         updateAnimal(data);
-
+        data.remove("Owner"); // EMC - Remove really old Owner data
         replaceUUIDString(data, "OwnerUUID", "Owner");
     }
 
diff --git a/ca/spottedleaf/dataconverter/minecraft/versions/V99.java b/ca/spottedleaf/dataconverter/minecraft/versions/V99.java
index 3a41b3498084eb58c97552f4ad4b869fa2d06438..2505269cdbc71bb4494b9a16ae6ca5676965adc2 100644
--- a/ca/spottedleaf/dataconverter/minecraft/versions/V99.java
+++ b/ca/spottedleaf/dataconverter/minecraft/versions/V99.java
@@ -358,6 +358,11 @@ public final class V99 {
                 if (tileTicks != null) {
                     for (int i = 0, len = tileTicks.size(); i < len; ++i) {
                         final MapType tileTick = tileTicks.getMap(i);
+                    // EMC start
+                    if (!tileTick.hasKey("p")) {
+                        tileTick.setInt("p", 0); // SavedTick priority NORMAL id.
+                    }
+                    // EMC end
                         WalkerUtils.convert(MCTypeRegistry.BLOCK_NAME, tileTick, "i", fromVersion, toVersion);
                     }
                 }
